File: python3/Plug.py
# -*- coding: utf-8 -*-

import re
from os import path,makedirs
from time import time

class Plug(object):
	def __init__(self,root_path, base_path='/'):
		self.root_path = root_path
		self.base_path = base_path
	
	def build(self,trigger,content,desc,scope='source.php'):
		tpl = '''<snippet>
	<content><![CDATA[%s]]></content>
	<scope>%s</scope>
	<tabTrigger>%s</tabTrigger>
	<description>%s</description>
</snippet>
		''' % (content,scope,trigger,desc)

		filename = re.sub(':','',trigger.strip() ).replace('\\','')
		# filename = re.sub('[\\/:]','',trigger.strip() )
		base_path = self.root_path + self.base_path
		filepath = base_path + filename + str( int(time()) ) + '.sublime-snippet'

		# 检测目录是否存在
		if not path.exists(base_path):
			makedirs(base_path)
		f = open(filepath, 'w')
		f.write(tpl)
		f.close()

if __name__ == '__main__':
  plug = Plug()
  # plug.build('where','where(...)','where条件','source.py')

File: php/plug.php
<?php

class Plug
{
	protected $root_path;
	protected $base_path='/';

	function __construct($root_path,$base_path)
	{
		$this->root_path = $root_path;
		$this->base_path = $base_path;
	}

	public function build($trigger,$content,$desc,$scope='source.php')
	{
		$tpl = "<snippet>
	<content><![CDATA[${content}]]></content>
	<scope>${scope}</scope>
	<tabTrigger>${trigger}</tabTrigger>
	<description>${desc}</description>
</snippet>";
		$filename = str_replace(':','',trim(strtolower($trigger)) );
		$base_path = $this->root_path.$this->base_path;
		$filepath = $base_path.$filename.time().'.sublime-snippet';

		is_dir($this->root_path) || mkdir($this->root_path);
		is_dir($base_path) || mkdir($base_path);

		file_put_contents($filepath, $tpl );
	}

}



// $plug = new Plug('C:\\Users\\Administrator\\Desktop\\1','\\');

// $plug->build('where','where()','where查询');